src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1242,19 +1246,191 @@
 
 
 (sc/defn to-beta-graph :- schema/BetaGraph
-
   "Produces a description of the beta network."
-  [productions :- #{schema/Production}]
-
-  (let [id-counter (atom 0)
-        create-id-fn (fn [] (swap! id-counter inc))]
-
-    (reduce (fn [beta-graph production]
+  [productions :- #{schema/Production}
+   create-id-fn :- IFn]
+  (reduce (fn [beta-graph production]
               (binding [*compile-ctx* {:production production}]
                 (add-production production beta-graph create-id-fn)))
 
             empty-beta-graph
-            productions)))
+            productions))
+
+(sc/defn extract-exprs :- {(schema/tuple sc/Int sc/Keyword) schema/SExpr}
```

Comment:
This schema is referred to a lot. It'd be helpful to extract it with a `s/defschema` so it is easier to read through the many places it is used.

Suggested `NodeExprFnLookup`